server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java

```diff
@@ -330,10 +330,10 @@ public HostForMigrationResponse newHostForMigrationResponse(HostJoinVO host, Enu
 
                 hostResponse.setHypervisorVersion(host.getHypervisorVersion());
 
-                Float cpuWithOverprovisioning = new Float(host.getCpus() * host.getSpeed() * ApiDBUtils.getCpuOverprovisioningFactor(host.getClusterId()));
-                String cpuAlloc = decimalFormat.format(((float)cpu / cpuWithOverprovisioning * 100f)).toString() + "%";
+                float cpuWithOverprovisioning = new Float(host.getCpus() * host.getSpeed() * ApiDBUtils.getCpuOverprovisioningFactor(host.getClusterId()));
+                String cpuAlloc = decimalFormat.format(((float)cpu / cpuWithOverprovisioning * 100f)) + "%";
```

Comment:
I see `100f` a couple of times on the code, there is also `100.0f`.

What do you think of extracting these _magic_ `100f`/`100.0f` into a constant?

Another approach that would be even nicer would be to have these pieces extracted to a method like `calculateResourceAllocatedPercentage`, and add a few unit test case methods.

```
public String calculateResourceAllocatedPercentage(float resource, float resourceWithOverprovisioning) {
	    return decimalFormat.format(((float)resource / resourceWithOverprovisioning * 100f)) + **"%"**
}
```

Reply:
@GabrielBrascher I made the changes. Also can you look into the issue #4221 ?

Reply:
@ravening thanks **+1** :)

Can you please take a look at line [314](https://github.com/apache/cloudstack/blob/7febf17aa8c8ba6403deeb160a4af5f6dda5111c/server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java#L314), I think that it could also benefit from the created method `calculateResourceAllocatedPercentage`.

Reply:
@GabrielBrascher I didn't make changes in line 314 because it has different behavior compared to line 165.
I have mentioned the same in issue #4221 also. list hosts api returns absolute value where as findhostsformigration returns percentage. If we decided what to return then I can make changes in these two places also

Reply:
@ravening sorry for the delay. I think that I was clear about my point.

- the method that you extract [L464](https://github.com/apache/cloudstack/blob/7febf17aa8c8ba6403deeb160a4af5f6dda5111c/server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java#L464):
`decimalFormat.format(((float)resource / resourceWithOverProvision * 100.0f)) + "%"`

- line [314](https://github.com/apache/cloudstack/blob/7febf17aa8c8ba6403deeb160a4af5f6dda5111c/server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java#L314) that I mentioned:
`decimalFormat.format((float) mem / memWithOverprovisioning * 100.0f) +"%"`

That is why I thought that it could also benefit from `calculateResourceAllocatedPercentage`.